app/services/match_orders.rb

```diff
@@ -30,7 +30,7 @@ def match_orders(sell_orders, buy_orders)
   private
 
   def orders_match?(buy_order, sell_order)
-    !sell_order.fulfilled &&
+    sell_order.fulfilled_at.nil? &&
```

Comment:
Maybe you could have a `fufilled?` method that does `fufilled_at.present?`

Would solve your test problem too.